app/helpers/application_helper.rb

```diff
@@ -23,14 +23,12 @@ def sent_requests(id)
   end
 
   def current_user_requests(id)
-    FriendRequest.exists?(user_id: current_user.id, friend_id: id)
+    current_user.friend_requests.exists?(friend_id: id)
   end
 
   def pending_requests(id)
-    request = FriendRequest.find_by(user_id: current_user.id, friend_id: id)
-    request_two = FriendRequest.find_by(user_id: id, friend_id: current_user.id)
-
-    request.nil? ? request_two.confirmed : request.confirmed
+    request = current_user.friends.find_by(user_id: id)
+    request.nil? ? true : false
```

Comment:
Use that to improve your code.

```ruby
# User model

# Users who requested to be friends (needed for notifications)
has_many :inverted_friendships, -> { where confirmed: false }, class_name: "FriendRequest", foreign_key: "friend_id"
has_many :friend_requests, through: :inverted_friendships
```

Reply:
Our helper method is just to check if a user is already a confirmed friend with another user.